### PR TITLE
[ENG-3958] - Add New Tests for User Settings Account Settings Page

### PR DESCRIPTION
**Pull request by DougCorell**
## Purpose
To expand selenium test coverage to include testing various aspects of the User Settings Account Settings Page.

## Summary of Changes

- .env.example - adding new environment variables for IMAP enabled email accounts.
- .github/workflows/nightly_core_functionality_tests.yml - adding new environment variables for IMAP enabled email accounts.
- .github/workflows/production_smoke_tests.yml - adding new environment variables for IMAP enabled email accounts.
- .github/workflows/weekly_regression_tests.yml - adding new environment variables for IMAP enabled email accounts.
- api/osf_api.py - adding 2 new functions: get_user_region_name() and get_region_data()
- components/email_access.py - new component that allows us to use IMAP enabled email account to retrieve and test emails sent from OSF.
- components/user.py - adding 5 new modal classes: ConfirmDeactivationRequestModal, UndoDeactivationRequestModal, Configure2FAModal, ConfirmEmailSentModal, and ConfirmRemoveEmailModal
- pages/user.py - adding new element locators to AccountSettingsPage page class
- settings.py - adding new environment variables for IMAP enabled email accounts.
- tests/test_user.py - adding 5 new tests: test_user_account_settings_connected_email(), test_user_account_settings_storage_locations(), test_user_account_settings_update_password(), test_user_account_settings_enable_2fa(), and test_user_account_settings_deactivate_account

## Reviewer's Actions
`git fetch <remote> pull/<PR_number>/head:testFix/user-account-settings`

Run this test using
`tests/test_user.py::TestUserAccountSettings -s -v`

## Testing Changes Moving Forward
N/A

## Ticket
ENG-3958: SEL: User Test - Add New Test for User Settings Account Settings Page
https://openscience.atlassian.net/browse/ENG-3958

### CODE REVIEW

File: tests/test_user.py [hunk 1]
```diff
@@ -111,6 +112,190 @@ def test_change_middle_name(self, driver, profile_settings_page, fake):
...
+                settings.IMAP_EMAIL
+            )
+            if unconfirmed_email is not None:
+                delete_button = unconfirmed_email.find_element_by_css_selector(
+                    'button[data-test-delete-button]'
+                )
+                delete_button.click()
+                # Verify email address in text of confirmation modal
+                assert (
+                    settings_page.confrim_remove_email_modal.deleted_email.text
```
Comment (felliott) [inline tests/test_user.py]: Please confrim the spelling of "confrim" :)

File: tests/test_user.py [hunk 2]
```diff
@@ -111,6 +112,190 @@ def test_change_middle_name(self, driver, profile_settings_page, fake):
...
+        # Click the listbox to display all of the storage locations and get a list
+        # of the displayed regions
+        settings_page.storage_location_listbox.click()
+        listbox_items = settings_page.storage_location_listbox.find_elements(
+            By.CSS_SELECTOR,
+            'div.ember-basic-dropdown-content-wormhole-origin > div > ul > li',
+        )
+        listbox_regions = []
+        for region in listbox_items:
+            listbox_regions.append(region.text)
```
Comment (felliott) [inline tests/test_user.py]: golf: `listbox_regions = [region.text for region in listbox_items]`

File: tests/test_user.py [hunk 3]
```diff
@@ -111,6 +112,190 @@ def test_change_middle_name(self, driver, profile_settings_page, fake):
...
+                    'Inbox',
+                    'SUBJECT',
+                    'Confirm account merge',
+                )
+                # Search through the email body and verify that the OSF account owner's
+                # email address is in the body of the email
+                assert str(email_body).find(settings.USER_ONE) > 0
+        finally:
+            # Lastly delete the unconfirmed email from the account
+            unconfirmed_email = settings_page.get_unconfimred_email_item(
```
Comment (felliott) [inline tests/test_user.py]: "unconfimred"

File: tests/test_user.py [hunk 4]
```diff
@@ -111,6 +112,190 @@ def test_change_middle_name(self, driver, profile_settings_page, fake):
...
+        settings_page.goto()
+        assert user.AccountSettingsPage(driver, verify=True)
+        # Scroll down to the Security settings section near the bottom of the page and
+        # click the Configure button
+        settings_page.scroll_into_view(settings_page.configure_2fa_button.element)
+        settings_page.configure_2fa_button.click()
+        # On the Configure 2FA Modal, first click the Cancel button and verify that the
+        # 2FA QR code is not yet displayed.
+        settings_page.configure_2fa_modal.cancel_button.click()
+        assert settings_page.two_facor_qr_code_img.absent()
```
Comment (felliott) [inline tests/test_user.py]: "facor"

File: tests/test_user.py [hunk 5]
```diff
@@ -111,6 +112,190 @@ def test_change_middle_name(self, driver, profile_settings_page, fake):
...
+        # of the displayed regions
+        settings_page.storage_location_listbox.click()
+        listbox_items = settings_page.storage_location_listbox.find_elements(
+            By.CSS_SELECTOR,
+            'div.ember-basic-dropdown-content-wormhole-origin > div > ul > li',
+        )
+        listbox_regions = []
+        for region in listbox_items:
+            listbox_regions.append(region.text)
+        listbox_regions.sort()
```
Comment (felliott) [inline tests/test_user.py]: golfier: `listbox_regions = sorted([region.text for region in listbox_items])`

File: tests/test_user.py [hunk 6]
```diff
@@ -111,6 +112,190 @@ def test_change_middle_name(self, driver, profile_settings_page, fake):
...
+        # Get available regions using the api and verify that the lists match
+        regions_data = osf_api.get_regions_data(session)
+        api_regions = []
+        for region in regions_data:
+            api_regions.append(region['attributes']['name'])
+        api_regions.sort()
```
Comment (felliott) [inline tests/test_user.py]: see prior golf

File: tests/test_user.py (hunk 1 above)
Comment (DougCorell) [reply tests/test_user.py]: D'oh!  - Fixed.

File: tests/test_user.py (hunk 2 above)
Comment (DougCorell) [reply tests/test_user.py]: Fixed.

File: tests/test_user.py (hunk 3 above)
Comment (DougCorell) [reply tests/test_user.py]: Double D'oh! - Fixed

File: tests/test_user.py (hunk 4 above)
Comment (DougCorell) [reply tests/test_user.py]: D'oh! D'oh! D'oh! - Fixed

File: tests/test_user.py (hunk 5 above)
Comment (DougCorell) [reply tests/test_user.py]: Fixed

File: tests/test_user.py (hunk 6 above)
Comment (DougCorell) [reply tests/test_user.py]: Fixed

**Comment by jh27539** [comment]
LGTM!

**Pull request merged**